__tests__/app.test.js

```diff
@@ -148,3 +148,61 @@ describe("GET /api/reviews", () => {
 			});
 	});
 });
+
+describe("POST /api/reviews/:review_id/comments. accepts an obj with username and body properties", () => {
+	test("POST 201 adds comment to database, returns comment obj", () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/1/comments")
+			.send(testComment)
+			.expect(201)
+			.then(({ body: { comment } }) => {
+				expect(comment.body).toBe("this review is so pointless!");
+				expect(comment.votes).toBe(0);
+				expect(comment.review_id).toBe(1);
+				expect(comment.comment_id).toBe(7);
+				expect(comment.author).toBe("bainesface");
+				expect(typeof comment.created_at).toBe("string");
+			});
+	});
+	test('POST 404- "sorry, review_id not found!"', () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/20/comments")
+			.send(testComment)
+			.expect(404)
+			.then((response) => {
+				expect(response.body.msg).toBe("sorry, review_id not found!");
+			});
+	});
+	test("POST 400- bad request!", () => {
+		const testComment = {
+			username: "bainesface",
+			body: "this review is so pointless!",
+		};
+		return request(app)
+			.post("/api/reviews/nonsense/comments")
+			.send(testComment)
+			.expect(400)
+			.then((response) => {
+				expect(response.body.msg).toBe("bad request!");
+			});
+	});
+	test("POST 400- Not Acceptable!", () => {
+		return request(app)
+			.post("/api/reviews/1/comments")
+			.send({})
+			.expect(400)
+			.then((response) => {
+				expect(response.body.msg).toBe(
+					"sorry, comment should be in the form of an obj with a username and a body property, both of which should be strings"
+				);
+			});
+	});
+});
```

Comment:
What happens if the username provided doesn't exist? Or if there are unnecessary properties in the sent object?

Reply:
We would need a test for if the posted username is incorrect (such as HannahBurks) this should throw a PSQL error due to breaking Foreign key rules. 

We also need a test for if unnecessary properties are added in the post, that it will ignore them.